### DISCUSSION TITLE
Request for Position: Remove support for data: URL in SVGUseElement

### DISCUSSION THREAD

**Discussion by uazo** [bromite/bromite#2495] [Q&A]
I got this [Request for Position: Remove support for data: URL in SVGUseElement](https://github.com/mozilla/standards-positions/issues/718) via subscriptions to that repo.
So I checked.

The thing seems, as I understand it, that it would be appropriate to [remove support](https://github.com/w3c/trusted-types/issues/357#issuecomment-1049761361) for url data: in the svg [use href](https://www.w3.org/TR/SVG2/struct.html#UseElement) tags.
It seems it could be exploited to [inject js code](https://portswigger-labs.net/xss/xss.php?x=%3Csvg%3E%3Cuse%20href%3D%22data%3Aimage%2Fsvg%2Bxml%2C%26lt%3Bsvg%20id%3D%27x%27%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%26gt%3B%26lt%3Bimage%20href%3D%271%27%20onerror%3D%27alert%281%29%27%20%2F%26gt%3B%26lt%3B%2Fsvg%26gt%3B%23x%22%20%2F%3E&context=html) and used in XSS attacks.
I checked and in chromium, the fetch cannot be cross-site, and, with [partitioned the blink cache](https://github.com/bromite/bromite/blob/master/build/patches/Partition-Blink-memory-cache.patch) and [fixed the access to cross-site blobs](https://github.com/bromite/bromite/blob/master/build/patches/Partition-blobs-by-top-frame-URL.patch), the data url: can (hopefully) only be from the same-site.
But
![image](https://user-images.githubusercontent.com/29201891/206858875-0bdb88b1-3596-44dd-afd2-95e464f227f7.png)
data url support is active and javascript working
[test](https://portswigger-labs.net/xss/xss.php?x=%3Csvg%3E%3Cuse%20href%3D%22data%3Aimage%2Fsvg%2Bxml%2C%26lt%3Bsvg%20id%3D%27x%27%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%26gt%3B%26lt%3Bimage%20href%3D%271%27%20onerror%3D%27alert%281%29%27%20%2F%26gt%3B%26lt%3B%2Fsvg%26gt%3B%23x%22%20%2F%3E&context=html) [source](https://portswigger.net/web-security/cross-site-scripting/cheat-sheet#data-url-with-use-element)

the point in the code where the [usercount](https://chromestatus.com/metrics/feature/timeline/popularity/4356) is counted is [SVGUseElement::UpdateTargetReference()](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/svg/svg_use_element.cc;l=210;drc=fbaec92ad5982f83aa4544d5c88d66d08034a9f4?q=SVGUseElement::UpdateTargetReference&ss=chromium%2Fchromium%2Fsrc) it is probably sufficient to remove the fetch if the data url is given.
what do you say?

**Comment by csagan5** [comment]
> it is probably sufficient to remove the fetch if the data url is given.

Why? Because the fetch is responsible for triggering `onerror` and other Javascript behaviours?

**Reply by uazo** [reply]
because the fetch takes the contents of the data url to be processed.
in the case of the image above, it contains the svg, the svg tries to load a file from the network, fails and launches the onerror in javascript

**Reply by csagan5** [reply]
By removing the fetch data URIs would still work but not able to load anything from network, correct?

**Reply by uazo** [reply]
> By removing the fetch data URIs would still work but not able to load anything from network, correct?

By removing the fetch, theoretically the href of the svg `use` [tag](https://www.w3.org/TR/SVG2/struct.html#UseElement) no longer accepts data urls. 
It should be checked anyway.
